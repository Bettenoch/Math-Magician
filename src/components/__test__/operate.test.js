import renderer from 'react-test-renderer';
import operate from '../../logic/operate';

describe('should match DOM snapshots', () => {
  it('should match the calculate DOM snapshot', () => {
    const tree = renderer.create(<operate />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('should perform given calculation', () => {
  test('should divide 25 by 5 to give 5', () => {
    const divide = operate('25', '5', 'รท');
    expect(divide).toEqual('5');
  });
  test('should add 25 by 5 to give 50', () => {
    const add = operate('25', '5', '+');
    expect(add).toEqual('30');
  });
  test('should subtract 25 by 5 to give 5', () => {
    const subtract = operate('25', '5', '-');
    expect(subtract).toEqual('20');
  });
  test('should multiply 5 by 5 to give 25', () => {
    const multiply = operate('5', '5', 'x');
    expect(multiply).toEqual('25');
  });
});
