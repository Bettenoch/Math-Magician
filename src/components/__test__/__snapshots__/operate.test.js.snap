// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match DOM snapshots should match the calculate DOM snapshot 1`] = `<operate />`;
